### PR TITLE
[ADD][10.0] stock_available_product_expiry - Add report

### PR DESCRIPTION
**Pull request by acsonefho**
**Description**
Add a report to have details about stock scrap expired

**Technically**

- [x] Add the report
- [x] Increase version

### CODE REVIEW

File: stock_available_product_expiry/reports/stock_scrap_expired.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                                    <div t-field="o.location_id.partner_id" t-options='{"widget": "contact", "fields": ["address", "name", "phone", "fax"], "no_marker": True, "phone_icons": True}'/>
+                                </div>
+                            </div>
+                        </div>
+                        <br/>
+                        <h1 t-field="o.name" class="mt0"/>
+                        <table class="table table-condensed">
+                            <thead>
+                                <tr>
+                                    <th t-if="o.origin"><strong>Order (Origin)</strong></th>
```
Comment (rousseldenis) [inline stock_available_product_expiry/reports/stock_scrap_expired.xml]: @acsonefho IMHO, should be 'Source Document' See https://github.com/OCA/stock-logistics-warehouse/blob/10.0/stock_available_product_expiry/models/stock_scrap_expired.py#L31

File: stock_available_product_expiry/reports/stock_scrap_expired.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                                        <span t-field="o.removal_date"/>
+                                    </td>
+                                </tr>
+                            </tbody>
+                        </table>
+                        <br/>
+                        <br/>
+                        <table class="table table-condensed" t-if="o.stock_scrap_expired_line_ids">
+                            <t t-set="lines" t-value="o.stock_scrap_expired_line_ids"/>
+                            <t t-set="has_barcode" t-value="any([move.product_id and move.product_id.sudo().barcode for move in lines.mapped('move_id')])"/>
```
Comment (rousseldenis) [inline stock_available_product_expiry/reports/stock_scrap_expired.xml]: @acsonefho Everyone has product access (no need of sudo). See https://github.com/odoo/odoo/blob/10.0/addons/product/security/ir.model.access.csv#L11

File: stock_available_product_expiry/reports/stock_scrap_expired.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                                        <t t-if="not line.product_id and line.package_id"><span  t-field="line.package_id"/></t></td>
+                                    <td class="text-right">
+                                        <span t-field="line.scrap_qty"/>
+                                    </td>
+                                    <td>
+                                        <span t-field="line.location_id"/>
+                                    </td>
+                                    <td t-if="has_barcode">
+                                        <span t-if="line.product_id.barcode">
+                                            <img t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s' % ('Code128', line.product_id.barcode, 600, 100)" style="width:300px;height:50px"/>
```
Comment (rousseldenis) [inline stock_available_product_expiry/reports/stock_scrap_expired.xml]: @acsonefho Should be EAN13. See https://github.com/odoo/odoo/blob/10.0/addons/product/report/product_product_templates.xml#L20

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some fixes

**Comment by acsonefho** [comment]
@rousseldenis It should be ok now :)

**Review by rousseldenis — APPROVED** [review]

**Review by Cedric-Pigeon — APPROVED** [review]

**Pull request merged**